fly/rc/targets.go

```diff
@@ -88,14 +88,20 @@ func SaveTarget(
 }
 
 func selectTarget(selectedTarget TargetName) (TargetProps, error) {
-	if selectedTarget == "" {
-		return TargetProps{}, ErrNoTargetSpecified
-	}
 	flyTargets, err := LoadTargets()
 	if err != nil {
 		return TargetProps{}, err
 	}
 
+	if selectedTarget == "" {
+		if len(flyTargets.Targets) != 1 {
+			return TargetProps{}, ErrNoTargetSpecified
```

Comment:
It might be worth giving this a distinct error message in the > 1 case. From a user's perspective "no target specified" would be confusing, since it was OK to not specify it until another target came around.

How about something like "no target specified and none available" for the 0 case, and "more than one target saved; target must be specified" for >1?

Reply:
yup, sounds like a much more helpful UX

Reply:
@vito  PTAL :)